lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -239,6 +242,21 @@ def add_metadata(doc, options)
         end
       end
 
+      def add_transaction_meta_data(doc, options = {})
+        transaction_meta_data = options[:transaction_meta_data] || []
+        return if transaction_meta_data.empty?
```

Comment:
Could be nice to simplify this into
`return unless (transaction_meta_data = options[:transaction_meta_data])`

Reply:
`add_transaction_meta_data ` is removed and `add_metadata` is reused.